### PR TITLE
feat: create profile page.

### PR DESCRIPTION
**Pull request by MjxOro**
## Summary
I created a profile page that checks the URL params of which user to load. If the username matches the current user, the UI will show the "edit settings" button which routes to the settings page. If the username in the URL params exists and does not match the current user, it will use the username to grab the information from the backend, store it in a redux state, and show a "follow button" or "unfollow button" depending if the user followed the queried user.  this handles an Axios "post" or "delete" request to follow or unfollow the user. For the UI,  I created a profile banner component that uses the current user and/or queried user as state to render the appropriate UI based on the data it takes. I also created a bannerbutton component that takes message(String), onClick(function), icon(Component), general(boolean), and isFollowing(boolean) props to render appropriate button to handle functionality and UI.

### Preview of the PR
- Current user page
profilepagecurrentuser
- Other user page
profilepageotheruser

### CODE REVIEW

File: src/components/ProfileBanner.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <div className="flex flex-col items-center min-h-[30%] justify-center px-[20%] w-full py-4 bg-tag-color">
+      <img
+        src={`${image}`}
+        className="min-w-20 min-h-20 w-[15%] h-[15%] max-w-2xl max-h-2xl rounded-full"
+        alt="Profile Banner Rounded Avatar"
+      />
+      <p className="my-4 font-bold text-xl">{username}</p>
+      {isMe ? (
+        <button
```
Comment (YoussefLaunchUp) [inline src/components/ProfileBanner.js]: Lets make each of these buttons their own component to help with readability and so we can use them in other places potentially.

File: src/components/ProfileBanner.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          type="button"
+          className="flex items-center self-end text-gray-400 hover:text-white border border-gray-400 hover:bg-gray-400 focus:ring-4 focus:outline-none focus:ring-gray-300 font-medium rounded-md text-sm px-5 py-2.5 text-center mr-2 mb-2"
+          onClick={handleClick}
+        >
+          <RiSettings3Fill />
+          <p>Edit Profile Settings</p>
+        </button>
+      ) : (
+        <button
+          type="button"
```
Comment (YoussefLaunchUp) [inline src/components/ProfileBanner.js]: If a user follows this profile this button should change to reflect that. (even if the user followed in a different session)

**Review by YoussefLaunchUp — CHANGES_REQUESTED** [review]
Great work, Follow system for profile needs a little change.

File: src/components/ProfileBanner.js (hunk 1 above)
Comment (MjxOro) [reply src/components/ProfileBanner.js]: done! created a component named BannerButton.

File: src/components/ProfileBanner.js (hunk 2 above)
Comment (MjxOro) [reply src/components/ProfileBanner.js]: done, refactored some code to handle the functionality and UI changes.

File: src/components/ProfileBanner.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import { useParams } from "react-router-dom";
+
+function ProfileBanner() {
+  let navitagion = useNavigate();
+  let dispatch = useDispatch();
+  let location = useLocation();
+  const { user:currentUser } = useSelector( state => state.auth );
+  const { profile } = useSelector(state => state.profile );
+  const { userId } = useParams();
+  useEffect( () =>{
```
Comment (YoussefLaunchUp) [inline src/components/ProfileBanner.js:19]: Nice use of the useEffect :)

**Review by YoussefLaunchUp — APPROVED** [review]
Great work!

**Pull request merged**